erpc/networks_bench_test.go

```diff
@@ -297,3 +297,109 @@ func BenchmarkNetworkForward_RetryFailures(b *testing.B) {
 		}
 	})
 }
+
+// BenchmarkNetworkForward_ConcurrentEthGetLogs tests performance when 500
+// concurrent clients call eth_getLogs on an upstream
+func BenchmarkNetworkForward_ConcurrentEthGetLogs(b *testing.B) {
+	util.ConfigureTestLogger()
+	util.ResetGock()
+	defer util.ResetGock()
+
+	gock.New("http://rpc-logs.localhost").
+		Persist().
+		Post("").
+		Reply(200).
+		JSON([]byte(`{"jsonrpc":"2.0","id":1,"result": []}`)) // empty logs array as result
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	rlr, err := upstream.NewRateLimitersRegistry(&common.RateLimiterConfig{
+		Budgets: []*common.RateLimitBudgetConfig{},
+	}, &log.Logger)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	mt := health.NewTracker("benchProject", 2*time.Second)
+	upConfig := &common.UpstreamConfig{
+		Type:     common.UpstreamTypeEvm,
+		Id:       "logs_upstream",
+		Endpoint: "http://rpc-logs.localhost",
+		Evm: &common.EvmUpstreamConfig{
+			ChainId: 123,
+		},
+	}
+
+	vr := thirdparty.NewVendorsRegistry()
+	pr, err := thirdparty.NewProvidersRegistry(&log.Logger, vr, []*common.ProviderConfig{}, nil)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	upsReg := upstream.NewUpstreamsRegistry(
+		ctx,
+		&log.Logger,
+		"benchProject",
+		[]*common.UpstreamConfig{upConfig},
+		rlr,
+		vr,
+		pr,
+		nil,
+		mt,
+		1*time.Second,
+	)
+	if err := upsReg.Bootstrap(ctx); err != nil {
+		b.Fatal(err)
+	}
+	if err := upsReg.PrepareUpstreamsForNetwork(ctx, util.EvmNetworkId(123)); err != nil {
+		b.Fatal(err)
+	}
+
+	ntw, err := NewNetwork(
+		ctx,
+		&log.Logger,
+		"benchProject",
+		&common.NetworkConfig{
+			Architecture: common.ArchitectureEvm,
+			Evm: &common.EvmNetworkConfig{
+				ChainId: 123,
+			},
+		},
+		rlr,
+		upsReg,
+		mt,
+	)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	upsList := ntw.upstreamsRegistry.GetNetworkUpstreams(util.EvmNetworkId(123))
+
+	upsList[0].EvmStatePoller().SuggestLatestBlock(0x11118000)
+	upsList[0].EvmStatePoller().SuggestFinalizedBlock(0x11117FFF)
+
+	requestBytes := []byte(`{
+      	"jsonrpc": "2.0",
+		"method": "eth_getLogs",
+		"params": [{
+			"fromBlock": "0x11118001",
+			"toBlock":   "0x11118888"
+		}],
+		"id": 1
+    }`)
+
+	b.SetParallelism(500)
+
+	b.ResetTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			fakeReq := common.NewNormalizedRequest(requestBytes)
+			_, err := ntw.Forward(ctx, fakeReq)
```

Comment:
This might not give us exact 500 concurrent requests based on GOMAXPROCS

So maybe having a for-loop around line 398 and trigger goroutines can make it happen?

I'm not fully sure though maybe `SetParallelism(500)` already gives us exactly what we need (i.e. in the logs you see ~500 multiplex requests)